content/docs/gke/deploy/deploy-cli.md

```diff
@@ -32,92 +32,130 @@ Before installing Kubeflow on the command line:
+    # If using basic authentication, create environment variables
+    # for username and password:
     export KUBEFLOW_USERNAME=<your username>
     export KUBEFLOW_PASSWORD=<your password>
-    ```
-
-1. Download a `kfctl` release from the 
-  [Kubeflow releases page](https://github.com/kubeflow/kubeflow/releases/).
-
-1. Unpack the tar ball:
-
-    ```
-    tar -xvf kfctl_<release tag>_<platform>.tar.gz
-    ```
-
-1. Run the following commands to set up and deploy Kubeflow. The code below
-  includes an optional command to add the binary `kfctl` to your path. If you 
-  don't add the binary to your path, you must use the full path to the `kfctl` 
-  binary each time you run it.
-
-    ```bash
-    # The following command is optional, to make kfctl binary easier to use.
-    export PATH=$PATH:<path to your kfctl file>
 
-    # Set KFAPP to the name of your Kubeflow application. See detailed
+    # Set KFAPP to the name of your Kubeflow application. This will be the
+    # name of the application directory too. See the detailed
     # description in the text below this code snippet.
-    # For example,  'kubeflow-test' or 'kfw-test'.
+    # For example, your app name can be  'kubeflow-test' or 'kfw-test'.
     export KFAPP=<your choice of application directory name>
 
-    export ZONE=<your target GCP zone> # where the deployment will be created
-    export PROJECT=<your GCP project ID>
+    # The following command is optional. It adds the kfctl binary to your path.
+    # If you don't add kfctl to your path, you must use the full path
+    # each time you run kfctl.
+    export PATH=$PATH:<path to your kfctl file>
+    ```
 
-    # Run the following commands for the default installation which uses Cloud IAP:
-    export CONFIG="{{% config-uri-gcp-iap %}}"
-    kfctl init ${KFAPP} --project=${PROJECT} --config=${CONFIG} -V
-    # Alternatively, run these commands if you want to use basic authentication:
-    export CONFIG="{{% config-uri-gcp-basic-auth %}}"
-    kfctl init ${KFAPP} --project=${PROJECT} --config=${CONFIG} -V --use_basic_auth
+<a id="set-up-and-deploy"></a>
+## Set up and deploy Kubeflow
+
+To set up and deploy Kubeflow using the **default settings**,
+run the **apply** command:
+
+```
+mkdir ${KFAPP}
+cd ${KFAPP}
+kfctl apply -V -f ${CONFIG}
+```
+
+Alternatively, you can set up your configuration files without deploying 
+Kubeflow, so that you can customize the configuration before deployment:
+
+1. Run the **build** command to set up your configuration:
+
+  ```
+  mkdir ${KFAPP}
+  cd ${KFAPP}
+  kfctl build -V -f ${CONFIG}
+  ```
+
+1. Edit the configuration files, as described in the guide to
+  [customizing your Kubeflow deployment](/docs/gke/customizing-gke).
+1. When you have finished editing the configuration, run the **apply**
+  command to deploy Kubeflow:
+
+  ```
+  kfctl apply -V -f ${CONFIG}
+  ```
+
+Notes:
+
+* **${KFAPP}** - The _name_ of a directory where you want Kubeflow 
```

Comment:
Name, Project, Zone, and Email can also be explicitly set by setting them in the config file.

If any of those values aren't set kfctl will attempt to get defaults as follows
gcloud for project, zone, and email (users can do gcloud config list to see the values).

NAME will be inferred from the directory where kfctl is run.

The following is a snippet that may be useful for helping people understand how to set them explicitly.

```
yq w -i kfctl_gcp_iap.yaml spec.plugins[0].spec.project ${PROJECT}
yq w -i kfctl_gcp_iap.yaml spec.plugins[0].spec.zone ${ZONE}
yq w -i kfctl_gcp_iap.yaml metadata.name ${KFNAME}
```

This is using yq from here.
https://github.com/mikefarah/yq/releases


* Project, Zone, and Email from the gcloud

Reply:
Thanks. Added.